apps/webcert/src/store/srs/srsMiddleware.tsx

```diff
@@ -281,11 +290,11 @@ export const handleUpdateCertificateDataElement: Middleware<Dispatch> =
   }
 
 const handleUpdateCertificate: Middleware<Dispatch> =
-  ({ dispatch }) =>
+  ({ dispatch, getState }) =>
   () =>
   (action: PayloadAction<Certificate>): void => {
     dispatch(resetState())
-    dispatch(updateUserClientContext(getUserClientContextForCertificate(action.payload.metadata)))
+    dispatch(updateUserClientContext(getUserClientContextForCertificate(action.payload.metadata, getState().ui.uiSRS.userLaunchFromOrigin)))
```

Comment:
Since ```dispatch(resetState())``` is called on the line before, the state has been resetted and userLaunchFromOrigin is empty.

Reply:
Should be fixed now